### ISSUE TITLE
Race conditions in subject can cause lock-ups at the start of periods

### ISSUE THREAD

**Issue by marmphco** [RedwoodAdmin/RedwoodFramework#44]
Occasionally a subject will receive a __page_loaded__ message from another subject before receiving __set_period__. This causes a deadlock in the _on_load synchronization barrier by preventing the _on_load barrier is never resolved. 

Normally, when a subject loads, it triggers _at_barrier. Upon receiving _at_barrier, a flag is set which marks the subject as having finished loading. When all the subjects have done this, the _on_load barrier is resolved and the period begins. 

However, when the message mis-ordering occurs, the _at_barrier message is triggered, but it is placed in the send queue rather than sent to the router. This is because _enable_messaging() is never called. Why isn't _enable_messaging() called? It turns out that this function is normally called after the subject has received __page_loaded__ from every other subject. The recv callback for __page_loaded__ returns if there is a mismatch in the period given by the message and the stored period for the subject that sent the message. So when __page_loaded__ is received before __set_period__, a period mismatch occurs and the subject never correctly marks all subjects as having loaded.

**TLDR: A bug that causes __page_loaded__ to be received before __set_period__ leads to frozen experiments.**

[Sample console output (with additional debug annotations)](https://gist.github.com/marmphco/df608a57ea0d6a3e56bb)
